src/sonic-ctrmgrd/ctrmgr/container_monitor.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+import signal
+import sys
+import syslog
+
+sys.path.append('/usr/share/sonic/scripts/')
+import container_state
+
+name = None
+
+def mark_end():
+    container_state.container_down(name)
+    syslog.syslog(syslog.LOG_INFO, "Marking end of docker {}".format(name))
+
+
+def handler(signum, frame):
+    syslog.syslog(syslog.LOG_ERR, "Received signal {}".format(signum))
+    mark_end()
+
+
+def handler(signum, frame):
+    syslog.syslog(syslog.LOG_ERR, "Received signal {}".format(signum))
+    mark_end()
+
+
+def main():
+    global name
+
+    if (len(sys.argv) < 2):
+        print("Usage {} <container name>".format(sys.argv[0]))
+        syslog.syslog(syslog.LOG_ERR, "Usage {} <container name>".format(sys.argv[0]))
+        sys.exit(-1)
+
+    name = sys.argv[1]
+    signal.signal(signal.SIGTERM, handler)
```

Comment:
if user is doing container kill, then this container_monitor will not receive SIGTERM signal, then it will not do proper clean up.

Reply:
Good point. Yes indeed. That is why `container kill` calls `container_state.down(feature)`. Comments are there in that code, explaining this.

Reply:
I removed the container_monitor.py completely, as the intended call upon receiving SIGTERM is not stable that there is a good potential of that script to get aborted. Moreover, when container crash exits or kubelet kills, there is no chance for termination program to be called. But what is *certain* to happen in any kind of container termination is call to "systemctl stop". The code to mark the feature/container down is now moved as part of "container stop" and tested.